routing/validation_barrier.go

```diff
@@ -3,12 +3,17 @@ package routing
 import (
 	"fmt"
 	"sync"
+	"time"
 
 	"github.com/lightningnetwork/lnd/channeldb"
 	"github.com/lightningnetwork/lnd/lnwire"
 	"github.com/lightningnetwork/lnd/routing/route"
 )
 
+// waitingParentSignalTimeout defines the time to wait before the dependant
+// gives up waiting for its parent's signal to either allow or deny.
+const waitingParentSignalTimeout = 60 * time.Second
```

Comment:
Make sense! I think it's better to open a new PR to have a more thorough discussion on this, will remove it since it doesn't fix the issue mentioned in the PR directly.